internal/pkg/runtime/engine/apptainer/container_linux.go

```diff
@@ -2804,7 +2804,7 @@ func (c *container) prepareNetworkSetup(system *mount.System, pid int) (func(con
 			if euid != 0 {
 				dropPrivilege, err := priv.Escalate()
 				if err != nil {
-					return err
+					return fmt.Errorf("while escalating the privilege of container process, err: %s, this is usually caused by not installing apptainer-suid", err)
```

Comment:
It shouldn't have been able to get that far.  There was already a check [earlier in that function](https://github.com/apptainer/apptainer/blob/release-1.4/internal/pkg/runtime/engine/apptainer/container_linux.go#L2758) that should have caught this condition.  It looks like there's a bug there, please investigate.

Reply:
thanks! I updated the code.